src/command/backup/backup.c

```diff
@@ -1439,23 +1440,23 @@ static ProtocolParallelJob *backupJobCallback(void *data, unsigned int clientIdx
 
                 // Create backup job
                 ProtocolCommand *command = protocolCommandNew(PROTOCOL_COMMAND_BACKUP_FILE);
-
-                protocolCommandParamAdd(command, VARSTR(manifestPathPg(file->name)));
-                protocolCommandParamAdd(
-                    command, VARBOOL(!strEq(file->name, STRDEF(MANIFEST_TARGET_PGDATA "/" PG_PATH_GLOBAL "/" PG_FILE_PGCONTROL))));
-                protocolCommandParamAdd(command, VARUINT64(file->size));
-                protocolCommandParamAdd(command, VARBOOL(!file->primary));
-                protocolCommandParamAdd(command, file->checksumSha1[0] != 0 ? VARSTRZ(file->checksumSha1) : NULL);
-                protocolCommandParamAdd(command, VARBOOL(file->checksumPage));
-                protocolCommandParamAdd(command, VARUINT64(jobData->lsnStart));
-                protocolCommandParamAdd(command, VARSTR(file->name));
-                protocolCommandParamAdd(command, VARBOOL(file->reference != NULL));
-                protocolCommandParamAdd(command, VARUINT(jobData->compressType));
-                protocolCommandParamAdd(command, VARINT(jobData->compressLevel));
-                protocolCommandParamAdd(command, VARSTR(jobData->backupLabel));
-                protocolCommandParamAdd(command, VARBOOL(jobData->delta));
-                protocolCommandParamAdd(command, VARUINT64(jobData->cipherType));
-                protocolCommandParamAdd(command, VARSTR(jobData->cipherSubPass));
+                PackWrite *param = protocolCommandParam(command);
```

Comment:
Why are archive push/get the only places where const is used: `PackWrite *const param = protocolCommandParam(command);` instead of `PackWrite *param = protocolCommandParam(command);`.  I'm wondering what the rule of thumb is here...

Reply:
They could all have the `*const`. Some of this code was written a while back before I started apply const to pointers like this. Likewise `PackWrite *commandParam = protocolCommandParam(command)` should get the `*const` and be named `param` to match better.

But having the `*const` or not does not change the functionality. It lets me  (and the compiler) know that the pointer won't be updated.

Reply:
Fixed in 195df7008dd6f2.